lib/doctest.js

```diff
@@ -459,98 +436,97 @@ function rewrite$coffee(options, input) {
 
-    return result.tests.map (function($test) {
+    return result.tests.map ($test => {
       normalizeTest ($test);
       return indentN ($test.indent.length, wrap$coffee ($test));
     });
   });
 
-  function append(s, line) { return s + line + '\n'; }
   return CoffeeScript.compile (
-    chunks.literalChunks.reduce (function(s, chunk, idx) {
-      return (testChunks[idx] || []).reduce (
-        append,
-        chunk.lines.reduce (append, s)
-      );
-    }, '')
+    chunks.literalChunks.reduce (
+      (s, chunk, idx) => Z.reduce (
+        (s, line) => `${s}${line}\n`,
+        chunk.lines.reduce ((s, line) => `${s}${line}\n`, s),
+        idx < testChunks.length ? testChunks[idx] : []
+      ),
+      ''
+    )
   );
-}
+};
 
-function functionEval(source) {
+const functionEval = source => {
   //  Functions created via the Function function are always run in the
   //  global context, which ensures that doctests can't access variables
   //  in _this_ context.
   //
   //  The `evaluate` function takes one argument, named `__doctest`.
-  var evaluate = Function ('__doctest', source);
-  var queue = [];
-  evaluate ({enqueue: function(io) { queue.push (io); }});
+  const evaluate = Function ('__doctest', source);
+  const queue = [];
+  evaluate ({enqueue: io => { queue.push (io); }});
   return run (queue);
-}
+};
 
-function commonjsEval(source, path) {
-  var abspath =
-  (pathlib.resolve (path)).replace (/[.][^.]+$/, '-' + Date.now () + '.js');
+const commonjsEval = (source, path) => {
+  const abspath = pathlib.resolve (path)
+                  .replace (/[.][^.]+$/, `-${Date.now ()}.js`);
 
   fs.writeFileSync (abspath, source);
-  var queue;
+  let queue;
   try {
     queue = (require (abspath)).__doctest.queue;
   } finally {
     fs.unlinkSync (abspath);
   }
   return run (queue);
-}
+};
 
-function run(queue) {
-  return queue.reduce (function(accum, io) {
-    var thunk = accum.thunk;
+const run = queue =>
+  queue.reduce ((accum, io) => {
+    const {thunk} = accum;
     if (io.type === INPUT) {
       if (thunk != null) thunk ();
       accum.thunk = io.thunk;
     } else if (io.type === OUTPUT) {
-      var either;
+      let either;
       try {
         either = {tag: 'Right', value: thunk ()};
       } catch (err) {
         either = {tag: 'Left', value: err};
       }
       accum.thunk = null;
-      var expected = io.thunk ();
+      const expected = io.thunk ();
 
-      var pass, repr;
+      let pass, repr;
       if (either.tag === 'Left') {
-        var name = either.value.name;
-        var message = either.value.message;
+        const {name, message} = either.value;
         pass = io['!'] &&
                name === expected.name &&
                message === expected.message.replace (/^$/, message);
-        repr = '! ' + name +
-               (expected.message && message.replace (/^(?!$)/, ': '));
+        repr = `! ${expected.message === '' ? name : either.value}`;
```

Comment:
When we want to use the `<name>: <message>` format we can simply use `` `! ${either.value}` ``, since `Error.prototype.toString` uses this format.